lib/Driver/ParseableOutput.cpp

```diff
@@ -180,24 +181,29 @@ class BeganMessage : public DetailedCommandBasedMessage {
 
 class TaskOutputMessage : public TaskBasedMessage {
   std::string Output;
+  Optional<sys::TaskFinishedMetrics> Metrics;
```

Comment:
You'll want to check with @graydon about overlap of functionality. He's already added much of this functionality in a different way.